tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/HttpClient/CurlHandler/CurlHandlerIntegration.cs

```diff
@@ -25,12 +25,11 @@ namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Http.HttpClient.CurlHand
         IntegrationName = IntegrationName)]
     [Browsable(false)]
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public class CurlHandlerIntegration
+    public unsafe class CurlHandlerIntegration
     {
         private const string IntegrationName = nameof(Configuration.IntegrationId.HttpMessageHandler);
         private const IntegrationId IntegrationId = Configuration.IntegrationId.HttpMessageHandler;
         private const IntegrationId CurlHandlerIntegrationId = IntegrationId.CurlHandler;
-        private static readonly Func<bool> IsIntegrationEnabledFunc = () => Tracer.Instance.Settings.IsIntegrationEnabled(CurlHandlerIntegrationId, defaultValue: true);
```

Comment:
Doesn't really matter, but in my telemetry branch, I've refactored the common Http stuff, and removed this func anyway: https://github.com/DataDog/dd-trace-dotnet/blob/andrew/telemetry/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/HttpClient/CurlHandler/CurlHandlerIntegration.cs

Reply:
Oh this is good to know, I'll try to find other cases for function pointers, if not, I will close this PR then. Thanks!

Reply:
I've revert the changes on Http integrations waiting for the telemetry PR in 11db0e4. But also I've added other integrations :)